### ISSUE TITLE
Users cannot log in after restoring iPhone from backup / transferring data to new device

### ISSUE THREAD

**Issue by mjmasn** [forcedotcom/SalesforceMobileSDK-iOS#3470]
Please fill out the following details:

1. Version of Mobile SDK Used: 9.2.0
2. Issue found in Native App or Hybrid App: React Native
3. OS Version: Various iOS 15.x
4. Device: Various iPhone
5. Steps to reproduce: Enable iCloud backup, wipe phone then restore from iCloud backup. Possibly applies to transferring data directly between devices too from colleague/customer reports.
6. Actual behavior: Error message after allowing login
![ec1cee6e-102f-4c8a-8a9d-db54f229688a](https://user-images.githubusercontent.com/2977095/149175141-29c250e4-0187-43cb-a58e-1663949ba6cb.png)

7. Expected Behavior: No error message after allowing login
8. Error Log: n/a

I've done some debugging but can't quite work out how to solve this. It looks like something is getting corrupted here such that `Encryptor.decrypt(data: encryptedKeyData, using: ecKeyPair(name: defaultKeyName).privateKey)` always throws: 

`Encryptor.swift`:
```
static func decrypt(data: Data, using key: SecKey) throws -> Data {
    var error: Unmanaged<CFError>?
    guard let decryptedData = SecKeyCreateDecryptedData(key, .eciesEncryptionStandardX963SHA256AESGCM, data as CFData, &error) else {
        let error = error?.takeRetainedValue()
        throw EncryptorError.decryptionFailed(underlyingError: error) // <--- throws here
    }
    return decryptedData as Data
}
```

`SFOAuthKeychainCredentials.m`: ignores the error
```
- (NSData *)encryptionKeyForService:(NSString *)service {
    NSData *keyForService = [SFSDKKeyGenerator encryptionKeyFor:service error:nil];
    return keyForService;
}
```

**Comment by mjmasn** [comment]
Just to clarify, this is not a recoverable error - I think the only solution would be to wipe the device completely and start from scratch (not from an icloud backup). Rebooting the device, uninstalling/reinstalling the app don't work.

**Comment by mjmasn** [comment]
Quick follow up - I reset a test device completely this morning without restoring from backup and the issue does not occur. So I think that confirms it's related to migration rather than an OS-version-specific issue.

**Comment by mjmasn** [comment]
PR linked above is the best solution we could come up with for now... I'm not an iOS developer though so may be better ways or other considerations. Seems to fix the issue for us though, have confirmed issue and fix multiple times on my iPhone 7 Plus and a colleague's iPad after restoring from icloud backups.

**Comment by bbirman** [comment]
Thanks for all the information and the proposed fix, we're looking into it

**Comment by bbirman** [comment]
Should be fixed by #3476 and will be in the 10.0 release. Thanks @mjmasn!
